### ISSUE TITLE
Dynamically building text blocks with set_fact no longer works

### ISSUE THREAD

**Issue by EmoBurrito** [ansible/ansible#39188]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ansible, set_fact

##### ANSIBLE VERSION

```
ansible 2.5.1
  config file = /home/username/redacted_path/ansible.cfg
  configured module search path = [u'/home/username/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### CONFIGURATION

N/A

##### OS / ENVIRONMENT

I would assume this is platform independent but I'm running from and targeting CentOS 7.4 boxes.

##### SUMMARY

I have list of dictionaries with IP's and hostnames in a role's `defaults` file that I use to dynamically build a block to be inserted into the target's hosts file. **This worked in Ansible 2.5.0 as expected.** After updating to 2.5.1, it seems to replace the previous entry rather than append to it, so we really only get the last entries.

##### STEPS TO REPRODUCE

`defaults/main.yml`
```yaml
---
# defaults file for hosts
host_block: ""
new_line: "\n"

hosts_users:
  - {ip: 192.168.10.1, name: tom}
  - {ip: 192.168.10.2, name: dick}
  - {ip: 192.168.10.3, name: harry}

hosts_servers:
  - {ip: 192.168.20.1, name: web}
  - {ip: 192.168.20.2, name: files}
  - {ip: 192.168.20.3, name: voip}
```

`tasks/main.yml`
```yaml
# This will append each item to the end of the currently existing block.
- name: Add user IP's to block
  set_fact:
    host_block: '{{ host_block }}{{ item.ip }}  {{ item.name }}.{{ domain_name }}.{{ domain_top_level }}  {{ item.name }}{{ new_line }}'
  with_items: '{{ hosts_users }}'

- name: Add server IP's to block
  set_fact:
    host_block: '{{ host_block }}{{ item.ip }}  {{ item.name }}.{{ domain_name }}.{{ domain_top_level }}  {{ item.name }}{{ new_line }}'
  with_items: '{{ hosts_servers }}'

- name: Check hosts block
  debug:
    msg: '{{ host_block }}'

- name: Add hosts
  become: yes
  blockinfile:
    path: /etc/hosts
    block: '{{ host_block }}'
    marker: "# {mark} CREATED BY ANSIBLE. Do not change."
```

##### EXPECTED RESULTS

Each IP would be added to the string in a format like:
```
IP  hostname.domainname.toplevel  hostname
```
This string would be output during the debug step so we can confirm it was properly built, and added as a block to the target's hosts file.

##### ACTUAL RESULTS

**Ansible 2.5.0 works as expected:**
```
TASK [hosts : Add user IP's to block] *************************************************************************
ok: [localhost] => (item={u'ip': u'192.168.10.1', u'name': u'tom'})
ok: [localhost] => (item={u'ip': u'192.168.10.2', u'name': u'dick'})
ok: [localhost] => (item={u'ip': u'192.168.10.4', u'name': u'harry'})

TASK [hosts : Add server IP's to block] *************************************************************************
ok: [localhost] => (item={u'ip': u'192.168.20.1', u'name': u'web'})
ok: [localhost] => (item={u'ip': u'192.168.20.2', u'name': u'files'})
ok: [localhost] => (item={u'ip': u'192.168.20.3', u'name': u'voip'})

TASK [hosts : Check hosts block] *************************************************************************
ok: [localhost] => {
    "msg": "192.168.10.1  tom.domain.toplevel  tom\n192.168.10.2  dick.domain.toplevel  dick\n192.168.10.3  harry.domain.toplevel  harry\n192.168.20.1  web.domain.toplevel  web\n192.168.20.2  files.domain.toplevel  files\n192.168.20.3  voip.domain.toplevel  voip\n"
}

TASK [hosts : Add hosts] *********************************************************************************
changed: [localhost]

```

Hosts file:
```
# BEGIN CREATED BY ANSIBLE. Do not change.
192.168.10.1  tom.domain.toplevel  tom
192.168.10.2  dick.domain.toplevel  dick
192.168.10.3  harry.domain.toplevel  harry
192.168.20.1  web.domain.toplevel  web
192.168.20.2  files.domain.toplevel  files
192.168.20.3  voip.domain.toplevel  voip
# END CREATED BY ANSIBLE. Do not change.
```

**Ansible 2.5.1 only has the last item in the list.** It iterates over each item but seems to be overwriting instead of appending.
```
TASK [hosts : Add user IP's to block] *************************************************************************
ok: [localhost] => (item={u'ip': u'192.168.10.1', u'name': u'tom'})
ok: [localhost] => (item={u'ip': u'192.168.10.2', u'name': u'dick'})
ok: [localhost] => (item={u'ip': u'192.168.10.4', u'name': u'harry'})

TASK [hosts : Add server IP's to block] *************************************************************************
ok: [localhost] => (item={u'ip': u'192.168.20.1', u'name': u'web'})
ok: [localhost] => (item={u'ip': u'192.168.20.2', u'name': u'files'})
ok: [localhost] => (item={u'ip': u'192.168.20.3', u'name': u'voip'})

TASK [hosts : Check hosts block] *************************************************************************
ok: [localhost] => {
    "msg": "192.168.10.3  harry.domain.toplevel  harry\n192.168.20.3  voip.domain.toplevel  voip\n"
}

TASK [hosts : Add hosts] *********************************************************************************
changed: [localhost]
```

Hosts file:
```
# BEGIN CREATED BY ANSIBLE. Do not change.
192.168.10.3  harry.domain.toplevel  harry
192.168.20.3  voip.domain.toplevel  voip
# END CREATED BY ANSIBLE. Do not change.
```

**Comment by sivel** [comment]
Duplicate of #38075 fixed in #38302

This fix will be included in v2.5.2

If you have further questions please stop by IRC or the mailing list:

   * IRC: #ansible on irc.freenode.net
   * mailing list: https://groups.google.com/forum/#!forum/ansible-project

**Comment by EmoBurrito** [comment]
My bad, sorry. Didn't see the other one. Thanks for the update.
